frontend/packages/topology/src/components/quick-search/QuickSearchModalBody.tsx

```diff
@@ -77,14 +86,11 @@ const QuickSearchModalBody: React.FC<QuickSearchModalBodyProps> = ({
   );
 
   const onEnter = React.useCallback(() => {
-    const viewAllLink = document.getElementById('viewAll');
-    const { activeElement } = document;
-    if (activeElement === viewAllLink) {
-      history.push(`/catalog/ns/${namespace}?keyword=${searchTerm}`);
-    } else if (selectedItem) {
-      history.push(selectedItem.cta.href);
+    if (selectedItem) {
+      const { href, callback } = selectedItem.cta;
+      callback ? callback() : history.push(href);
     }
-  }, [selectedItem, namespace, searchTerm]);
+  }, [selectedItem]);
```

Comment:
Minor issue is that the clicking `enter` while focused on the view all links causes the user to navigate to the selected item instead of the view all link.
I think this can be fixed by stopping propagation of the event on the view all `Link`.

Reply:
I think the stop propagation won't work here because of the way event listener is added for enter key. We'll have to add a check that distinguishes a keyDown on view all link vs the selectedItem. Maybe we can merge this PR and handle that separately considering its the last day of this sprint. @christianvogt WDYT?

Reply:
`onEnter` should probably only be executed when the text input has focus.

Reply:
@divyanshiGupta please create an issue / PR to followup and address this.